language-design.md

```diff
@@ -17,6 +17,22 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+1. "Does this mean, then, that it is of no use to design? Not at all. But in stead of
+designing a thing, you need to design a way of doing. And this way of doing must make
+some choices now but leave other choices to a later time." [Steele, 1998]
+
+I liked this quote for a couple reasons. First, it seemed like an encapsulation of a lot of what Steele was trying to say in his talk; in designing a language, one shouldn't think necessarily about exactly what the users of the language need. Rather, according to Steele, one should think about how to allow the users to make the things they need and expand the language themselves. I think this is an interesting and important point, and the quote chosen describes that really well. I also liked the quote because it seems relevant to more than just programming language design. I think it applies to life in a way, and the notion that oftentimes, the process and learning gleaned from doing a task is as important (or more so) than the task itself.
+
+2. Steele describes introducing Java as it is to a new programmer audience, saying: "Programmers would have cried, “Too big! Too much hair! I can’t deal with all that!” But in real life it has worked out fine because the users have grown with the language and learned it piece by piece, and they buy in to it because they have had some say in how to change the language" [Steele, 1998, p. 14]. In practice, we found that the second part of this claim is certainly true. Historically, to learn Java as it was developed was to learn it through experience, piece by piece. Like Steele mentioned, learning this way is natural, and models the way we actually program. At the same time though, now that Java has been fully developed and flushed out, people do need to learn it in its current size. This leads to the interesting conclusion that learning a new language that is already "large" by Steele's definition must be done by learning it piece by piece, as if it was in the process of being developed. Shoving an entire language on someone at once would be an overwhelming way to learn it. 
+
```

Comment:
Interesting point, and I agree with most of what you've said here.  But don't we learn all languages pieces by piece, regardless of size?  If a language is medium-sized we'd learn it piece by piece, there'd just be fewer pieces to learn than there are with large languages.  Think about Context Free Art.  Quite a small language compared to the likes of Java, but we still learn it one step at a time.  So is there really anything different (other than the time it takes) about the process of learning different sized languages?